internal/proxyservice/tcp/mysql/protocol/protocol.go

```diff
@@ -781,6 +794,34 @@ func CheckPacketLength(expected int, packet []byte) error {
 	return nil
 }
 
+// ScrambleSHA256Password hashes password using MySQL 8+ method (SHA256)
+func ScrambleSHA256Password(scramble []byte, password string) []byte {
+	if len(password) == 0 {
+		return nil
+	}
+
+	// XOR(SHA256(password), SHA256(SHA256(SHA256(password)), scramble))
+
+	crypt := sha256.New()
+	crypt.Write([]byte(password))
+	message1 := crypt.Sum(nil)
+
+	crypt.Reset()
+	crypt.Write(message1)
+	message1Hash := crypt.Sum(nil)
+
+	crypt.Reset()
+	crypt.Write(message1Hash)
+	crypt.Write(scramble)
+	message2 := crypt.Sum(nil)
+
+	for i := range message1 {
+		message1[i] ^= message2[i]
+	}
+
+	return message1
+}
+
```

Comment:
What the reason we didn't need this before?

Reply:
Taken out and addressed in another PR. This is for expanded support.